geometry_msgs/Pose pose
bool use_plane_constraint